test/test_main.cpp

```diff
@@ -1,9 +1,55 @@
 #define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
 #include "catch.hpp"
 
-// cppcheck-suppress unusedFunction
-TEST_CASE("Example Test Case") {
-    // cppcheck-suppress knownConditionTrueFalse
-    // cppcheck-suppress duplicateExpression
-    REQUIRE(10 == 10);
+class ObstacleDetection {
```

Comment:
#include "your_class.hpp"
"""
TEST_CASE("{INSERT SHORT TEST DESCRIPTION}", "{INSERT TAGS}") {
    // Preconditions:
    {INSERT CODE THAT IS REQUIRED TO BE RUN BEFORE TESTING}

    // Run Test:
    {INSERT TESTED CODE}

    // Compare Postconditions:
    REQUIRE({COMPUTED VALUE} {OPERATOR} {CONSTANT VALUE});
}
"""